src/runner.rs

```diff
@@ -338,10 +338,11 @@ impl Interpreter {
 		instructions: &isa::Instructions,
 	) -> Result<RunResult, TrapKind> {
 		let mut iter = instructions.iterate_from(function_context.position);
+
 		loop {
-			let instruction = iter.next().expect("instruction");
+			let instruction = iter.next().expect("Function ended without explicit return");
```

Comment:
Yeah it's somewhat more complicated: it's not only an explicit return that can return from a function but also an implicit `end`. Moreover, function execution can also be terminated by a trap.

How about "Due to validation: it's impossible to reach the end of function."?